solr/core/src/java/org/apache/solr/cloud/overseer/ReplicaMutator.java

```diff
@@ -252,8 +254,7 @@ private ZkWriteCommand updateState(final ClusterState prevState, ZkNodeProps mes
       log.info("Failed to update state because the replica does not exist, {}", message);
       return ZkStateWriter.NO_OP;
     }
-    boolean persistCollectionState = collection != null && collection.isPerReplicaState();
-
+    boolean persistPrsCollectionState = false;
```

Comment:
I personally found this `persistPrsCollectionState` flag quite hard to follow:
1. Many branches and flipping in various places
2. Confusing for non PRS collection too, as this flag could be confusing for dev debugging code and their non PRS collection (they might not even be aware of what PRS is) has this flag set to `true` 

By looking at the original design of this method `updateState` - the behavior is well defined - if someone invoke this method, it would expect a single write to `state.json` (ie return `new ZkWriteCommand(collectionName, newCollection);`)

However, we are making special cases for PRS here, and I am wondering if it's truly necessary - if it's not an issue for non PRS collection, perhaps it's not really an issue for PRS too? This should not raise any race condition (as it's writing to state.json only) unless data node still has code that updates state.json too? There were some places that did ZKStateWriter.setData directly for PRS collection and I think we should fix those instead

Reply:
This can be simplified. let me do it anyway